### PR TITLE
feat: Allow .Release.Name to be used in gotmpl values templates

### PR DESCRIPTION
**Pull request by joshes**
This adds the ability to utilize `.Release` inside of gotmpl files as discussed [here](https://github.com/roboll/helmfile/issues/760).

Resolves: https://github.com/roboll/helmfile/issues/760

Added a simple test that passed once implemented and all tests are running green.

### CODE REVIEW

**Comment by Romiko** [comment]
Hi, is this something that will be merged in soon?

**Comment by mumoshu** [comment]
@Romiko Yes. I'll merge this today or tomorrow, immediately after I've finished stabilizing the latest minor version of Helmfile (v0.125.x).

@joshes Thanks for your contribution! This LGTM overall. But please let me modify this before/after merging this a bit, so that it doesn't accidentally expose other ReleaseSpec fields.

For example, this change as its current form result in exposing `Release.Values` to the template. It sounds useful initially, but in reality, it can contain either paths to values file OR map[string]interface{} of embedded values, which makes it hard for people to actually use. I'd like to discuss about it in a dedicated feature request(if someone submits it). Same for other ReleaseSpec fields.

**Comment by joshes** [comment]
@mumoshu Sounds good. That side effect did strike me as extremely useful as well initially but I hear what you're saying. :)

Thanks!

File: pkg/state/types.go [hunk 1]
```diff
@@ -19,12 +19,28 @@ type EnvironmentTemplateData struct {
 	Values map[string]interface{}
 }
 
-// releaseTemplateData provides variables accessible while executing golang text/template expressions in releases of a helmfile YAML file
+// releaseTemplateData provides variables accessible while executing golang text/template expressions in release templates
+// and release values templates within a Helmfile YAML file.
 type releaseTemplateData struct {
```
Comment (mumoshu) [inline pkg/state/types.go:24]: @joshes I realized that we already have a struct type for the template data used for rendering anything related to each release! So I refactored your change without modifying EnvironmentTemplateData.

Along the way I also realized that we already and unexpectedly exposed ReleaseSpec to the whole template in some case.

Fortunately, Helmfile had unit tests([`TestHelmState_executeTemplates`](https://github.com/roboll/helmfile/blob/3f02b866401ba7ef34c4c985de13151f3e51fb7b/pkg/state/state_exec_tmpl_test.go#L25)) that helped me to figure out which ReleaseSpec fields are intended to be exposed to the template. I extracted the fields to releseTemplateDataRelease(L34-L46).

**Review by mumoshu — APPROVED** [review]
@joshes Thanks a lot for implementing this! I especially appreciate the addition to the unit test, which helped me easily refactor this ☺️

**Pull request merged**